simulator/src/simulator/simulator.ts

```diff
@@ -46,27 +49,26 @@ export class Simulator {
   /** Number of nodes that have been started */
   nodeCount = 0
 
-  constructor(logger: Logger, options?: { persist?: boolean; duration?: number }) {
+  constructor(
+    logger: Logger,
+    options?: { persist?: boolean; duration?: number; verboseLogging?: boolean },
+  ) {
     this.logger = logger
     this.logger.withTag('simulator')
 
     this.running = true
 
-    if (options) {
-      const { persist, duration } = options
-
-      if (persist) {
-        this.persistNodeDataDirs = true
-      }
-
-      if (duration) {
-        this.logger.log(`this simulation will run for ${duration} seconds`)
-        const exitTimeout = setTimeout(() => {
-          this.logger.log(`simulator exiting after ${duration} seconds`)
-          clearTimeout(exitTimeout)
-          this.exit()
-        }, duration * SECOND)
-      }
+    this.verboseLogging = !!options?.verboseLogging
```

Comment:
why is it `!!`?

Reply:
To convert to a boolean if it's undefined